scipy/spatial/tests/test_kdtree.py

```diff
@@ -1224,7 +1224,7 @@ def test_ckdtree_weights():
         w1 = weights.copy()
         w1[i] = 0
         data2 = data[w1 != 0]
-        w2 = weights[w1 != 0]
+        weights[w1 != 0]
```

Comment:
Minor comment--I couldn't see a role for this statement at all & when I delete it locally the kdtree test suite still passes.

I suppose one might argue that the statement might have *some* value in that it could raise an exception if something is truly wrong, though if that was the purpose there probably should be a comment about it.

Reply:
Yes, some unused assignments of an expressions should still have the expression evaluated, this is not one of them.  Removed.